src/System.Tuples/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -0,0 +1,1119 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+
+namespace System
+{
+    /// <summary>
+    /// Helper so we can call some tuple methods recursively without knowing the underlying types.
+    /// </summary>
+    internal interface ITupleInternal
+    {
+        int GetHashCode(IEqualityComparer comparer);
+        int Size { get; }
+    }
+
+    public struct ValueTuple : IEquatable<ValueTuple>, IStructuralEquatable, IStructuralComparable, IComparable, ITupleInternal
+    {
+        public override bool Equals(object obj)
+        {
+            return ((IStructuralEquatable)this).Equals(obj, EqualityComparer<object>.Default);
+        }
+
+        public bool Equals(ValueTuple other)
+        {
+            return true;
+        }
+
+        bool IStructuralEquatable.Equals(object other, IEqualityComparer comparer)
+        {
+            return other is ValueTuple;
+        }
+
+        int IComparable.CompareTo(object obj)
+        {
+            return ((IStructuralComparable)this).CompareTo(obj, Comparer<object>.Default);
+        }
+
+        int IStructuralComparable.CompareTo(object other, IComparer comparer)
+        {
+            if (other == null) return 1;
+
+            if (!(other is ValueTuple))
+            {
+                throw new ArgumentException(SR.ArgumentException_ValueTupleIncorrectType, nameof(other));
+            }
+
+            // Note that the result doesn't depend on comparer
+            return 0;
+        }
+
+        public override int GetHashCode()
+        {
+            return ((IStructuralEquatable)this).GetHashCode(EqualityComparer<object>.Default);
```

Comment:
There's no need to box here.  It could instead be:
```C#
return EqualityComparer<object>.Default.GetHashCode();
```
or if you decide to change the hash code based on my comment below, just:
```C#
return 0;
```